packages/std/src/query/bank.rs

```diff
@@ -21,6 +24,10 @@ pub enum BankQuery {
     /// Note that this may be much more expensive than Balance and should be avoided if possible.
     /// Return value is AllBalanceResponse.
     AllBalances { address: String },
+    /// This calls into the native bank module for querying metadata for all bank tokens.
+    /// Return value is AllDenomMetadataResponse
+    #[cfg(feature = "cosmwasm_1_3")]
+    AllDenomMetadata {},
```

Comment:
Without pagination/limit this can be an expensive call. I checked with Osmosis and they have >970 entries already.
Can you elaborate on your use case a bit more? 

Would it make sense to add a query for a single metadata, too? it seems to be low hanging fruits for me.

Reply:
I think if only the available Metadata is returned, and not all the denoms that do not have a metadata entry, we don't have this sort of issue. "for all bank tokens" is then misleading and we need to clarify what the caller can expect.

I'm also strongly in favour us adding the query with one denom too. This can serve many use cases more efficiently than the list.

Reply:
For our use case, we only need to query for all denoms. That being said, I'm happy to add the query for one denom as well so that others who need it can start using it without another cosmwasm/wasmd/wasmvm upgrade.

@webmaster128 is correct that not all denoms have a metadata entry. However, I agree with @alpe that we shouldn't make this assumption for all chains and it makes sense to add limits/pagination to the query. 

I will update the comments accordingly to reflect the fact that this query is not "for all bank tokens"; it's "for all bank tokens that have a metadata entry".